manual/manual/cmds/comp.etex

```diff
@@ -358,6 +358,37 @@ command line, and possibly the "-custom" option.
 
 This section describes and explains in detail some warnings:
 
+\subsection{ss:warn6}{Warning 6: Label omitted in function application}
+
+OCaml supports "labels-omitted" full applications: if the function has
+a known arity, all the arguments are unlabeled, and their number
+matches the number of non-optional parameters, then labels are ignored
+and non-optional parameters are matched in their definition
+order. Optional arguments are defaulted.
+
+\begin{verbatim}
+let f ~x ~y = x + y;;
+let test = f 2 3;;
+\end{verbatim}
+
+This support for "labels-omitted" application was introduced when
+labels were added to OCaml, to ease the progressive introduction of
+labels in a codebase. However, it has the downside of weakening the
+labeling discipline: if you use labels to prevent callers from
+mistakenly reordering two parameters of the same type, labels-omitted
+make this mistake possible again.
+
+Warning 6 warns when labels-omitted applications are used, to
+discourage their use. When labels were introduced, this warning was
+not enabled by default, so users would use labels-omitted
+applications, often without noticing.
+
+Over time, most OCaml codebases adopted this warning to avoid
```

Comment:
Do you have a statistical analysis to show that or is that an unfounded guess? I do believe that this is true, but this sentence announce this point as a fact.

Reply:
"default in dune" => essentially all dune-projects use it. I believe that a majority of the open-source OCaml code being written today uses dune as a build system, but this is just a guess.

Reply:
(The guess will become even more true once the PR is merged!)

Reply:
Would it work to switch to "it became idiomatic to activate this warning" since we have a clear example of idiom using this warning?

Reply:
It is certainly true that a majority of new opam packages being submitted use dune nowadays. There may be large OCaml monorepos around still (e.g.) using omake, but I'm not aware of any that interact with opam repository and/or are open source.